ffi/src/handle.rs

```diff
@@ -367,8 +366,8 @@ mod private {
         type Raw = T;
 
         fn into_handle_ptr(val: Arc<T>) -> NonNull<T> {
-            let val = ManuallyDrop::new(val);
-            val.as_ref().into()
+            let ptr = Arc::into_raw(val);
+            NonNull::new(ptr as *mut T).expect("into_raw should be non-null")
```

Comment:
Ahh yeah, nice find. I was wondering about the tagging with Arc.

It _feels_ like in the arc case we'd actually _like_ it if the pointer could be `SharedReadOnly` as that's the semantics for Arc and it would allow cross language checking with something like CHERI if it understood the same tag semantics across languages.

For now though, `Arc::from_raw` seems to want it tagged `SharedReadWrite`. I'll do a bit more testing to make sure that's the case though.